src/GitVersion.Core/VersionCalculation/TrunkBased/NonTrunk/CommitOnNonTrunkWithPreReleaseTagBase.cs

```diff
@@ -6,7 +6,7 @@ namespace GitVersion.VersionCalculation.TrunkBased.NonTrunk;
 internal abstract class CommitOnNonTrunkWithPreReleaseTagBase : ITrunkBasedIncrementer
 {
     public virtual bool MatchPrecondition(TrunkBasedIteration iteration, TrunkBasedCommit commit, TrunkBasedContext context)
-        => commit.ChildIteration is null && !commit.Configuration.IsMainline
+        => !commit.HasChildIteration && !commit.Configuration.IsMainline
```

Comment:
```suggestion
        => !commit.HasChildIteration && !commit.Configuration.IsTrunkBased
```

Reply:
The IsMainline property is part of the branch related configuration system. If I rename it it would be effect also the github and gitflow workflow. I think IsMainline can reused for the trunk-based workflow as well to identify the trunk.

Reply:
How does `IsMainline` affect the GitHub and Git Flow workflows? I would prefer consistent naming here, as everywhere else `Mainline` has been renamed to `TrunkBased`.

Reply:
If I rename it to is-trunk (not trunk-based. trunk-based is a workflow) the configuration will look like:

![image](https://github.com/GitTools/GitVersion/assets/56404113/11824f18-f13f-48bd-a547-070c81775950)

Reply:
And even if you create an additional property: From the semantic point of view the IsMainline property can be reused. Because this is the main branch which will be the trunk in trunkbased workflow. If you mix trunk based with other strategies then it would be very strange to have two properties. I don't want to bring this complexity into the system because I don't see the use cases (KIS?). To have a mainline branch but not a trunk branch or otherway around would be error prone and confusing.

```yaml
  main:
    label: ''
    increment: Patch
    prevent-increment-of-merged-branch-version: true
    track-merge-target: false
    regex: ^master$|^main$
    source-branches:
    - develop
    - release
    tracks-release-branches: false
    is-release-branch: false
    is-trunk: true
    is-mainline: false
    pre-release-weight: 55000
```

Reply:
I think two concepts might be conflated here. `Mainline` is the name of a workflow mode that we want to remove. That is a separate concept from what is considered the "main" branch (usually named `main` or `master`). Which of these concepts does `IsMainline` and `is-mainline` represent?

Reply:
> Mainline is the name of a workflow mode that we want to remove.

Yes exactly I want to delete the implementation of Mainline and replace it with another. I decided to name it TrunkBasedStrategy and not MainlineStrategy to separate it from the IsMainline property to minimize the confusion.

> Which of these concepts does IsMainline and is-mainline represent?

IsMainline describes just the characteristics of a branch. In TrunkBasedStrategy it will be used and in TaggedCommitVersionStrategy it will be also used. We have other properties which will be used in other Strategies as well. To separate it makes no sense except you have a dedicated use case. Can you give me an example where we have a trunk and not a mainline or other way around? There is no use cases if you ask me because trunk and mainline is the same from the semantic point of view (main and support branch in all workflows).

With all respect I think it is out of scope of this PR because IsMainline was used in Mainline mode previously. If you like you can create a ticket and improve this. No problem.

Reply:
If I understand you correctly, `IsMainline` and `is-mainline` declares whether a branch is to be considered the "main" branch or not. Then the word `line` should be eradicated from the word, and `IsMainBranch` and `is-main-branch` are much more precise names for the property.

Reply:
Yes I is-main would be more precise. As soon we have only one property I agree.